modules/rosetta/src/main/scala/org/tessellation/rosetta/http/routes/ConstructionRoutes.scala

```diff
@@ -36,6 +34,14 @@ final class ConstructionRoutes[F[_]: Async](
           .asRosettaResponse
           .handleUnknownError
       }
+
+    case req @ POST -> Root / "preprocess" =>
+      req.decodeRosettaWithNetworkValidation[ConstructionPreprocess.Request](appEnvironment, _.networkIdentifier) { preprocessReq =>
+        constructionService
+          .getAccountIdentifiers(preprocessReq.operations)
+          .map(accountIds => ConstructionPreprocess.Response(Some(accountIds)))
```

Comment:
The type here is `Option[ConstructionPreprocess.Response]`. How do I convert that to `F[Response[F]]` to then apply `.asRosettaResponse.handleUnknownError`?

Reply:
so, in this case there is no error returned by the service (as the function is pure) so you can forget about the error and just return 200 Ok

```scala
{ preprocessReq =>
  val accountIdentifiers = constructionService.getAccountIdentifiers(preprocessReq.operations)
  
  Ok(ConstructionPreprocess.Response(accountIdentifiers))
}
```
or
```scala
{ preprocessReq =>
  val accountIdentifiers = constructionService.getAccountIdentifiers(preprocessReq.operations)
  
  Ok(ConstructionPreprocess.Response(accountIdentifiers))
    .handleUnknownError
}
```

Reply:
I cannot confirm it right now, but if above would not be sufficient (by returning `Response[F]` instead of `F[Reponse[F]]`), you can always wrap it into `F` by `.pure[F]` (from `Applicative`)